bin/dart_language_server.dart

```diff
@@ -17,11 +18,13 @@ Future main(List<String> args) async {
       var shim = await startShimmedServer(startupArgs);
       await new StdIOLanguageServer.start(shim).onDone;
     } catch (e, st) {
-      await new File('/tmp/lsp-error.log').writeAsString('Caught $e\n$st');
+      await new File(path.join(Directory.systemTemp.path, 'lsp-error.txt'))
+          .writeAsString('Caught $e\n$st');
     } finally {
       await closeLogs();
     }
   }, onError: (e, st) {
-    new File('/tmp/lsp-error.log').writeAsString('UnCaught $e\n$st');
+    new File(path.join(Directory.systemTemp.path, 'lsp-error.txt'))
```

Comment:
Not sure if this should be writeAsStringSync or should be awaited? Probably Dart won't quit while it's outstanding, but nodejs makes it easy to throw away pending writes when the process exits!